packages/api/src/users/users.controller.spec.ts

```diff
@@ -0,0 +1,100 @@
+import { BadRequestException, INestApplication, NotFoundException } from '@nestjs/common';
+import { CqrsModule } from '@nestjs/cqrs';
+import { Test } from '@nestjs/testing';
+
+import type { UserSurveyDTO } from '../../../shared/src/models/user';
+import { UserRepositoryService } from '../contracts/user.repository.service';
+import { MemoryDbService } from '../memoryDB/memoryDB.service';
+import { PgMemUserRepositoryAdapter } from '../memoryDB/user.repository.service';
+import { UserFactory } from './user.factory';
+import { UsersController } from './users.controller';
+import { CommandHandlers, EventHandlers } from './users.module';
+import { UsersRepository } from './users.repository';
+import { UsersService } from './users.service';
+
+describe('Users controller', () => {
+  let app: INestApplication;
+  let usersController: UsersController;
+  let userSurvey: UserSurveyDTO;
+
+  describe('saveUserSurvey', () => {
+    it('Should update user and create UserSurvey', async () => {
+      expect(await usersController.saveUserSurvey(userSurvey)).toBeTruthy();
+    });
+
+    it('Should throw not found error if user does not exists', async () => {
+      await expect(usersController.saveUserSurvey({ ...userSurvey, id: 99 })).rejects.toBeInstanceOf(NotFoundException);
+    });
+
+    it('Should throw bad request error if survey already exists', async () => {
+      await expect(usersController.saveUserSurvey(userSurvey)).rejects.toBeInstanceOf(BadRequestException);
+    });
+  });
+
+  async function setup() {
+    const module = await Test.createTestingModule({
+      imports: [CqrsModule],
+      controllers: [UsersController],
+      providers: [
+        MemoryDbService,
+        {
+          provide: UserRepositoryService,
+          useClass: PgMemUserRepositoryAdapter,
+        },
+        UserFactory,
+        UsersRepository,
+        UsersService,
+        ...CommandHandlers,
+        ...EventHandlers,
+      ],
+    }).compile();
+
+    app = module.createNestApplication();
+    usersController = app.get<UsersController>(UsersController);
+
+    const db = app.get<MemoryDbService>(MemoryDbService);
+    const repository = app.get<UserRepositoryService>(UserRepositoryService);
+
+    await Promise.all([app.init(), db.migrate()]);
+    await repository.create({
+      githubId: 12345678,
+      fullName: 'Name',
+      email: 'example@test.com',
+      image: 'https://photo-url.com',
+    });
+
+    userSurvey = {
```

Comment:
I recommend get rid of survey references from users module at all.
We need two-way relation. Leave just reference by userId from Survey to User. 
There is no business rule between user data like email / gender and the survey, so we can split it.

If we go this way, we will end up references to all other modules in users and users test, because almost everything in the system is related to some user. The test will blow up with things totally unrelated to users and any changes / new features requires changes in users module with is totally oposite of Open-Closed.